app/components/Onboarding/Steps/ConnectionDetails.js

```diff
@@ -207,36 +218,56 @@ class ConnectionDetails extends React.Component {
                   required
                 />
               </Box>
-
-              <Box mb={3}>
-                <Input
-                  field="connectionCert"
-                  name="connectionCert"
-                  label={<FormattedMessage {...messages.cert_title} />}
-                  description={<FormattedMessage {...messages.cert_description} />}
-                  initialValue={connectionCert}
-                  onValueChange={this.handleConnectionCertChange}
-                  validateOnBlur={shouldValidateInline}
-                  validateOnChange={shouldValidateInline}
-                  asyncValidate={this.validateCert}
-                  required
-                />
-              </Box>
-
-              <Box mb={3}>
-                <Input
-                  field="connectionMacaroon"
-                  name="connectionMacaroon"
-                  label="Macaroon"
-                  description={<FormattedMessage {...messages.macaroon_description} />}
-                  initialValue={connectionMacaroon}
-                  onValueChange={this.handleConnectionMacaroonChange}
-                  validateOnBlur={shouldValidateInline}
-                  validateOnChange={shouldValidateInline}
-                  asyncValidate={this.validateMacaroon}
-                  required
-                />
-              </Box>
+              <WithSelectDir
+                render={({ openDialog }) => (
+                  <Flex mb={3} flexDirection="row" alignItems="center">
+                    <Input
+                      field="connectionCert"
+                      name="connectionCert"
+                      width={1}
+                      label={<FormattedMessage {...messages.cert_title} />}
+                      description={<FormattedMessage {...messages.cert_description} />}
+                      initialValue={connectionCert}
+                      onValueChange={this.handleConnectionCertChange}
+                      validateOnBlur={shouldValidateInline}
+                      validateOnChange={shouldValidateInline}
+                      asyncValidate={this.validateCert}
+                      required
+                    />
+                    <SelectFileButton
+                      onClick={e => {
+                        e.preventDefault()
```

Comment:
If `SelectFileButton` button type was `button` (see previous comment), this line can probably be removed.

Reply:
OK i'll check that